src/components/results/use-results-tab.ts

```diff
@@ -38,11 +38,85 @@ export type ResultTabIndexRedirection =
  */
 export const useResultsTab = (
     resultTabIndexRedirection: ResultTabIndexRedirection,
-    setTabIndex: (index: number) => void,
-    tabLevel: ResultsTabsLevel
-) => {
+    setTabIndex: React.Dispatch<React.SetStateAction<number>>,
+    view: string
+): Dispatch<SetStateAction<Boolean>> => {
+    const [redirectionLock, setRedirectionLock] =
+        useResultsTabRedirectionLock();
+
+    useEffect(() => {
+        if (view !== StudyView.RESULTS && !redirectionLock) {
+            setTabIndex(resultTabIndexRedirection);
+        }
+    }, [resultTabIndexRedirection, setTabIndex, view, redirectionLock]);
+
+    return setRedirectionLock;
+};
+
+const useResultsTabRedirectionLock = (): [
+    Boolean,
+    Dispatch<SetStateAction<Boolean>>
+] => {
+    const computationStatus: ComputingStatus = useSelector(
+        (state: ReduxState) => state.computingStatus
+    );
+    const [redirectionLock, setRedirectionLock] = useState<Boolean>(false);
+
+    //we ought to release the redirection lock if the user launch a new computation
     useEffect(() => {
-        resultTabIndexRedirection?.[tabLevel] &&
-            setTabIndex(resultTabIndexRedirection[tabLevel]);
-    }, [tabLevel, resultTabIndexRedirection, setTabIndex]);
+        setRedirectionLock(false);
+    }, [computationStatus]);
+
+    return [redirectionLock, setRedirectionLock];
+};
+
+const getServiceResultTabIndex = (
+    computingType: ComputingType,
+    availableServices?: IService[]
+) => {
+    return availableServices
+        ? availableServices
+              .map((service) => service.computingType)
+              .findIndex((computations) => computations.includes(computingType))
+        : ResultsTabsRootLevel.LOADFLOW;
+};
+
+//function to convert computing type to the index corresponding to the resulting computation tab.
+//the list of services must also be passed as parameter because its content is dynamic depending on the state of the application (e.g. developer mode disabled)
+export const computingTypeToRootTabRedirection = (
+    computingType: ComputingType,
+    availableServices?: IService[]
+): ResultTabIndexRedirection => {
+    console.log(computingType);
+
+    const rootResultTabIndex = getServiceResultTabIndex(
+        computingType,
+        availableServices
+    );
+    switch (computingType) {
+        case ComputingType.LOADFLOW:
+        case ComputingType.SECURITY_ANALYSIS:
+        case ComputingType.SENSITIVITY_ANALYSIS:
+        case ComputingType.DYNAMIC_SIMULATION:
+        case ComputingType.VOLTAGE_INIT:
+        case ComputingType.ALL_BUSES_SHORTCIRCUIT_ANALYSIS:
+        case ComputingType.ONE_BUS_SHORTCIRCUIT_ANALYSIS:
+        case ComputingType.NON_EVACUATED_ENERGY_ANALYSIS:
+            return rootResultTabIndex;
+        default:
+            return ResultsTabsRootLevel.LOADFLOW;
+    }
+};
```

Comment:
From what I understand, this would work as well and would be easier to understand
I don't understand this switch role

```suggestion
//function to convert computing type to the index corresponding to the resulting computation tab.
//the list of services must also be passed as parameter because its content is dynamic depending on the state of the application (e.g. developer mode disabled)
export const computingTypeToRootTabRedirection = (
    computingType: ComputingType,
    availableServices?: IService[]
): ResultTabIndexRedirection => {
    return availableServices
        ? availableServices
              .map((service) => service.computingType)
              .findIndex((computations) => computations.includes(computingType))
        : ResultsTabsRootLevel.LOADFLOW;
};
```

Reply:
The switch was originally useful because the specific redirection related to shortcircuit sub computations were also handled here but since I moved them out of it you're right it's now useless